service/builder.go

```diff
@@ -121,40 +142,85 @@ func (b *builder) addDescriptor(prefix string, consumes []string, produces []str
 	}
 }
 
+// copyDefinition creates a deep copy from original definition. The copy not very deep for those
+// fields which type is interface{}.
 func (b *builder) copyDefinition(d *definition.Definition, consumes []string, produces []string) *definition.Definition {
-	// It copy fields except document.
-	newOne := &definition.Definition{}
-	*newOne = *d
-	if d.Consumes != nil {
-		newOne.Consumes = make([]string, len(d.Consumes))
-		copy(newOne.Consumes, d.Consumes)
-	} else if consumes != nil {
-		newOne.Consumes = make([]string, len(consumes))
-		copy(newOne.Consumes, consumes)
+	newOne := &definition.Definition{
+		Method:      d.Method,
+		Function:    d.Function,
+		Description: d.Description,
+	}
+	if len(d.Consumes) > 0 {
+		consumes = d.Consumes
 	}
-	if d.Produces != nil {
-		newOne.Produces = make([]string, len(d.Produces))
-		copy(newOne.Produces, d.Produces)
-	} else if produces != nil {
-		newOne.Produces = make([]string, len(produces))
-		copy(newOne.Produces, produces)
+	newOne.Consumes = make([]string, len(consumes))
+	copy(newOne.Consumes, consumes)
+
+	if len(d.Produces) > 0 {
+		produces = d.Produces
+	}
+	newOne.Produces = make([]string, len(produces))
+	copy(newOne.Produces, produces)
+
+	if len(d.ErrorProduces) > 0 {
+		produces = d.ErrorProduces
```

Comment:
It means we use specified `ErrorProduces` rather than inherit from `Produces`.  If you want to use   produces, you can simply leave `ErrorProduces` empty.